include/swift/AST/DiagnosticsSema.def

```diff
@@ -3975,8 +3975,8 @@ ERROR(attr_incompatible_with_objc,none,
       (DeclAttribute, DescriptiveDeclKind))
 
 ERROR(final_not_on_accessors,none,
-      "'final' cannot be applied to accessors, it must be put on the "
-      "%select{var|let|subscript}0", (unsigned))
+      "'final' cannot be applied to %1, it must be put on the "
+      "%select{var|let|subscript}0", (unsigned, DescriptiveDeclKind))
```

Comment:
I think this could be argued either way depending on a personal preference just like the rest of the points. If more than one accessor there declares ‘final’ we still get a diagnostic per use, so using a kind vs. ‘accessors’ seems like a moot point for clarity.

Reply:
In this spirit this particular diagnostic could be changed into something like - `only {property, subscript} itself can be marked as  ‘final’` and attach it to the parent declaration with note and fix-it per ‘final’ use.

Reply:
> so using a kind vs. ‘accessors’ seems like a moot point for clarity.

The stress is on the volume of educational information that we can offer without losing clarity, rather than clarity itself. If we can equip the user with the knowledge to not make a similar mistake, like putting `final` on a setter this time, as in `class can only inherit from another class` vs. `class cannot inherit from struct 'S'`, I cannot think why that wouldn’t be a better user experience.

But anyway, your suggestion sounds fine for a compromise 🙂 `'final' must be put on the {var, subscript} instead` is another possibility, but I think I like yours better.

Reply:
I understand what you are saying but when it comes to accessors at least there are ways to rephrase these messages that don’t involve terminology and don’t loose quality. 

When it comes to the message you referenced I think it could say `… cannot inherit from struct …` and put additional information into a note to expand what is possible if users need that.

Reply:
I think it all depends on what is really emphasized by a diagnostic…

Reply:
> When it comes to the message you referenced I think it could say `… cannot inherit from struct …` and put additional information into a note to expand what is possible

Mm, this did not occur to me. I love it!